src/experimental/Toast/ToastUI.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react'
+import {CheckIcon, AlertIcon, StopIcon, InfoIcon, XIcon} from '@primer/octicons-react'
+
+import {Spinner, Box, IconButton} from '../..'
+
+const variants = ['default', 'success', 'attention', 'danger', 'loading'] as const
+
+const icon: Record<ToastVariants, React.ReactNode> = {
+  default: <InfoIcon size="small" />,
+  success: <CheckIcon size="small" />,
+  attention: <AlertIcon size="small" />,
+  danger: <StopIcon size="small" />,
+  loading: <Spinner size="small" />,
+}
+
+export const ToastUI = ({
+  dismissable = false,
+  variant = 'default',
+  role = 'status',
+  dismissLabel = 'Dismiss this message',
+  children,
+}: React.PropsWithChildren<ToastUIProps>) => {
+  if (__DEV__) {
+    // eslint-disable-next-line no-console
+    console.assert(
+      variants.includes(variant),
+      `The variant ${variant} is not supported. Use one of ${variants.join(', ')}`,
+    )
+  }
+
+  return (
+    <Box role={role} sx={{display: 'flex', justifyContent: 'flex-start'}}>
+      <Box
+        sx={{display: 'inline-flex', maxWidth: '450px', borderRadius: 2, boxShadow: 'shadow.large', overflow: 'hidden'}}
```

Comment:
```suggestion
        sx={{display: 'inline-flex', maxWidth: 'min(100vw, 450px)', borderRadius: 2, boxShadow: 'shadow.large', overflow: 'hidden'}}
```

I _think_ we might want to have all toasts be a min width of 450px on larger screens even if they have a small amount of content. I'm imagining if multiple toasts appear, we would want them to all have the same width in a stack. We might want to discuss this further in a pattern working group session as well with other designers. What do you think?

I'm pretty sure the CSS I suggested will give in a min width but also let it scale down for mobile 🤞

Reply:
I'd love to see if @maximedegreve has an opinion here! Should the toast width respond to the side of the contents, or should it have a min width so they stack evenly? I have seen it done both ways and I don't have a strong opinion.

Reply:
Hi, yeah thanks! What im wanting to achieve here is a maxWidth, hence the property. But assigned to the minimum value between 100vw and 450px. I guess this could have been articulated through `width: 100%; max-width: 45px` as well.

Reply:
Haha you're playing with my emotions! I quickly mocked a few scenarios up and I'm leaning to not use a max-width on desktop breakpoints. On mobile breakpoints I would always set it to `100vw`.

Those are hot takes though I gave this only 10 minutes of thoughts and maybe there are more arguments than the ones described below.

![Future](https://github.com/primer/react/assets/980622/30419829-4cd2-492c-97f7-738ffae38c47)

Reply:
@maximedegreve my gut reaction is that the Toasts that respond to the content size look more natural compared to having a fixed width, so I agree with you! Thanks for those mockups ❤️ 

I don't know if we've designed for where they will appear (center top or bottom left) so that might change things if its not centered, but I think we should start with this design and go from there.